test/services/AuthenticatorServiceSpec.scala

```diff
@@ -0,0 +1,107 @@
+ * limitations under the License.
+ */
+
+package services
+
+import connectors.mocks.MockAuthenticatorConnector
+import helpers.TestConstants._
+import org.scalatest.matchers.must.Matchers
+import org.scalatest.wordspec.AnyWordSpec
+import uk.gov.hmrc.http.HeaderCarrier
+import uk.gov.hmrc.soletraderidentificationfrontend.models.SoleTraderDetailsMatching.{Deceased, Matched, Mismatch, NotFound}
+import uk.gov.hmrc.soletraderidentificationfrontend.services.AuthenticatorService
+import play.api.test.Helpers.{await, defaultAwaitTimeout}
+
+import scala.concurrent.ExecutionContext.Implicits.global
+import scala.concurrent.Future
+
+class AuthenticatorServiceSpec extends AnyWordSpec with Matchers with MockAuthenticatorConnector {
+
+  object TestService extends AuthenticatorService(mockAuthenticatorConnector)
+
+  implicit val hc: HeaderCarrier = HeaderCarrier()
+
+  "matchSoleTraderDetails" should {
+    "return Right(Matched)" when {
+      "the provided details match those from authenticator" when {
+        "the enableSautrCheck is true and the sautr is provided" in {
+          mockMatchSoleTraderDetails(testSoleTraderDetails)(Future.successful(Right(testSoleTraderDetails)))
+
+          val result = await(TestService.matchSoleTraderDetails(testSoleTraderDetails, testJourneyConfig(enableSautrCheck = true)))
+
+          result mustBe Right(Matched)
+        }
+
+        "the enableSautrCheck is false and the sautr is not provided" in {
+          mockMatchSoleTraderDetails(testSoleTraderDetailsNoSautr)(Future.successful(Right(testSoleTraderDetailsNoSautr)))
+
+          val result = await(TestService.matchSoleTraderDetails(testSoleTraderDetailsNoSautr, testJourneyConfig()))
+
+          result mustBe Right(Matched)
+        }
+      }
+    }
+
+    "return Left(Mismatch)" when {
+      "the provided details do not match those from authenticator" when {
+        "the enableSautrCheck is true and the sautr is provided" in {
+          mockMatchSoleTraderDetails(testSoleTraderDetails)(Future.successful(Left(Mismatch)))
+
+          val result = await(TestService.matchSoleTraderDetails(testSoleTraderDetails, testJourneyConfig(enableSautrCheck = true)))
+
+          result mustBe Left(Mismatch)
+        }
+
+        "the enableSautrCheck is false and the sautr is not provided" in {
+          mockMatchSoleTraderDetails(testSoleTraderDetailsNoSautr)(Future.successful(Left(Mismatch)))
+
+          val result = await(TestService.matchSoleTraderDetails(testSoleTraderDetailsNoSautr, testJourneyConfig()))
+
+          result mustBe Left(Mismatch)
+        }
+      }
+
+      "the provided sautr does not exist on authenticator" in {
+        mockMatchSoleTraderDetails(testSoleTraderDetails)(Future.successful(Right(testSoleTraderDetailsNoSautr)))
+
+        val result = await(TestService.matchSoleTraderDetails(testSoleTraderDetails, testJourneyConfig(enableSautrCheck = true)))
+
+        result mustBe Left(Mismatch)
+      }
+    }
```

Comment:
Could we add a test the other way, where the user provides no SAUTR but there is one on authenticator?